src/components/Newsfeed.jsx

```diff
@@ -26,24 +31,98 @@ export const Newsfeed = () =>
         }
         else
         {
-            getAllArticles()
+            getAllArticles(queryString)
             .then(({articles}) =>
             {
                 setArticles(articles)
                 setLoading(false);
             });
         }
-    }, [isLoading])
+    }, [isLoading, searchParams])
+
+    const validSortQueries = ["article_id", "title", "topic", "author", "body", "created_at", "votes"];
```

Comment:
this validation is really necessary as it should be already dealt with in the backend so if any bad request come through the backend will catch it first